File: apps/api/BHMHockey.Api.Tests/SampleTest.cs
using FluentAssertions;
using Xunit;

namespace BHMHockey.Api.Tests;

/// <summary>
/// Sample test to verify the test framework is set up correctly.
/// Delete this file after adding real tests.
/// </summary>
public class SampleTest
{
    [Fact]
    public void TestFramework_IsConfiguredCorrectly()
    {
        // Arrange
        var expected = 4;

        // Act
        var result = 2 + 2;

        // Assert (using FluentAssertions)
        result.Should().Be(expected);
    }

    [Theory]
    [InlineData(1, 2, 3)]
    [InlineData(5, 5, 10)]
    [InlineData(-1, 1, 0)]
    public void TestFramework_TheoryTests_Work(int a, int b, int expected)
    {
        // Act
        var result = a + b;

        // Assert
        result.Should().Be(expected);
    }
}

File: packages/api-client/__tests__/sample.test.ts
/**
 * Sample test to verify Jest is configured correctly.
 * Delete this file after adding real tests.
 */
describe('API Client Test Framework Setup', () => {
  it('should run basic assertions', () => {
    expect(2 + 2).toBe(4);
  });

  it('should handle objects', () => {
    const config = {
      baseURL: 'http://localhost:5001/api',
      timeout: 5000,
    };
    expect(config).toHaveProperty('baseURL');
    expect(config.timeout).toBe(5000);
  });

  it('should handle promises', async () => {
    const mockApiCall = (): Promise<{ success: boolean }> => {
      return Promise.resolve({ success: true });
    };

    const result = await mockApiCall();
    expect(result.success).toBe(true);
  });
});
